ddtrace/tracer/writer.go

```diff
@@ -0,0 +1,277 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"math"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+type traceWriter interface {
+	// add adds a trace to the current payload being constructed by the handler.
+	add([]*span)
+
+	// flush causes the handler to send its current payload. The flush can happen asynchronously, but
+	// the handler must be ready to accept new traces with add when flush returns.
+	flush()
+
+	// stop shuts down the traceWriter, ensuring all payloads are flushed.
+	stop()
+}
+
+type agentTraceWriter struct {
+	config *config
+
+	payload *payload
+
+	// climit limits the number of concurrent outgoing connections
+	climit chan struct{}
+
+	// wg waits for all goroutines to exit when stopping.
+	wg sync.WaitGroup
+
+	// prioritySampling refers to the tracer's priority sampler.
```

Comment:
It does not. It can refer to anything a user sets it to. There is nothing stopping anyone from setting it to any value. For this reason, it is always best to look at each structure in isolation, completely separate from the rest of the code and project.
```suggestion
	// prioritySampling specifies the sampler to read rates into when sent as an HTTP response
	// from the agent.
```